Client/game_sa/CVehicleSA.cpp

```diff
@@ -684,6 +684,17 @@ void CVehicleSA::RemoveVehicleUpgrade(DWORD dwModelID)
         push    dwModelID
         call    dwFunc
     }
+
+    // GTA SA only does this when CVehicle::ClearVehicleUpgradeFlags returns false.
+    // In the case of hydraulics and nitro, this function does not return false and the upgrade is never removed from the array
+    for (auto& upgrade : GetVehicleInterface()->m_upgrades)
```

Comment:
Please, do not use `auto` instead of short typenames.

Reply:
If possible, also use `const`

Reply:
This cannot be const because the value is changed to -1

Reply:
Didnt see that, github mobile truncates the file to only changed lines